x/twap/types/errors.go

```diff
@@ -24,3 +24,32 @@ func (e StartTimeAfterEndTimeError) Error() string {
 	return fmt.Sprintf("called GetArithmeticTwap with a start time that is after the end time."+
 		" (start time %s, end time %s)", e.StartTime, e.EndTime)
 }
+
+type KeySeparatorLengthError struct {
```

Comment:
I feel like these typed errors in most cases isn't a great use of focus

Not opposed to it, but would not prioritize continuing it. (Lots of code / extra thought, for edge cases we likely won't hit after initial writing)

I feel like the approach for improving debugging should be "hit a bad debugging situation, then make a generalized solution to a large class of problems"

I feel like I mostly care about:
* stacktrace / line error was emitted at
* 'wrapped' context for how I got here.

Focus on error types at the bottom isn't how we achieve either.